src/main/java/com/nutomic/syncthingandroid/fragments/SelectFolderFragment.java

```diff
@@ -0,0 +1,75 @@
+package com.nutomic.syncthingandroid.fragments;
+
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.util.Log;
+import android.view.View;
+import android.widget.AdapterView;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+
+public class SelectFolderFragment extends FoldersFragment {
+
+    private static final String TAG = "SelectFolderFragment";
+
+    @Override
+    public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
+        for (Uri uri : files) {
+            File sourceFile = new File(getRealPathFromURI(uri));
+            String path = mAdapter.getItem(i).path;
+            File destFile = new File(path, sourceFile.getName());
+            try {
+                copy(sourceFile, destFile);
+            }
+            catch (IOException e) {
+                Log.e(TAG, "Failed copying file", e);
+            }
+        }
+
+        getActivity().finish();
+    }
+
+    public void setFiles(ArrayList<Uri> files) {
```

Comment:
Use of non-default constructors for Fragments is discouraged and bound to cause trouble. The official way to pass data and have them persisted for the lifetime of your fragment is by getting a reference to the Fragment and adding stuff to the arguments field. Use setArguments and getArguments and update the fields as needed. 
You should probably wrap the specifics inside the fragment so that callers don't have to worry about argument keys.

To get a reference fragment, check the fragment manager's find methods. There are many ways.